ts/src/pro/xt.ts

```diff
@@ -0,0 +1,1079 @@
+         * @param {object} params extra parameters specific to the xt api
+         * @returns {object} data from the websocket stream
+         */
+        await this.loadMarkets ();
+        const privateAccess = access === 'private';
+        let type = undefined;
+        [ type, params ] = this.handleMarketTypeAndParams (methodName, market, params);
+        const isContract = (type !== 'spot');
+        const subscribe = {
+            'method': isContract ? 'SUBSCRIBE' : 'subscribe',
+            'id': this.numberToString (this.milliseconds ()) + name,  // call back ID
+        };
+        if (privateAccess) {
+            if (!isContract) {
+                subscribe['params'] = [ name ];
+                subscribe['listenKey'] = await this.getListenKey (isContract);
+            } else {
+                const listenKey = await this.getListenKey (isContract);
+                const param = name + '@' + listenKey;
+                subscribe['params'] = [ param ];
+            }
+        } else {
+            subscribe['params'] = [ name ];
+        }
+        const tradeType = isContract ? 'contract' : 'spot';
+        const messageHash = name + ':' + tradeType;
+        const request = this.extend (subscribe, params);
+        let tail = access;
+        if (isContract) {
+            tail = privateAccess ? 'user' : 'market';
+        }
+        const url = this.urls['api']['ws'][tradeType] + '/' + tail;
+        return await this.watch (url, messageHash, request, messageHash);
+    }
+
+    async watchTicker (symbol: string, params = {}) {
+        /**
+         * @method
+         * @name xt#watchTicker
+         * @description watches a price ticker, a statistical calculation with the information calculated over the past 24 hours for a specific market
+         * @see https://doc.xt.com/#websocket_publictickerRealTime
+         * @see https://doc.xt.com/#futures_market_websocket_v2tickerRealTime
+         * @see https://doc.xt.com/#futures_market_websocket_v2aggTickerRealTime
+         * @param {string} symbol unified symbol of the market to fetch the ticker for
+         * @param {object} params extra parameters specific to the xt api endpoint
+         * @param {string} params.method 'agg_ticker' (contract only) or 'ticker', default = 'ticker' - the endpoint that will be streamed
+         * @returns {object} a [ticker structure]{@link https://docs.ccxt.com/en/latest/manual.html#ticker-structure}
+         */
+        const market = this.market (symbol);
+        const options = this.safeValue (this.options, 'watchTicker');
+        const defaultMethod = this.safeString (options, 'method', 'ticker');
+        const method = this.safeString (params, 'method', defaultMethod);
+        const name = method + '@' + market['id'];
+        return await this.subscribe (name, 'public', 'watchTicker', market, params);
+    }
+
+    async watchTickers (symbols: string[] = undefined, params = {}) {
```

Comment:
If we don't use the `symbols` how can we properly handle scenarios like this:
```Javascript
const promiseA = await xt.watchTickers(['BTC/USDT', 'LTC/USDT])
const promiseB = await xt.watchTickers(['ADA/USDT', 'SOL/USDT])
```

Reply:
> If we don't use the `symbols` how can we properly handle scenarios like this:
> 
> ```js
> const promiseA = await xt.watchTickers(['BTC/USDT', 'LTC/USDT])
> const promiseB = await xt.watchTickers(['ADA/USDT', 'SOL/USDT])
> ```

There's no way to do that, the exchange will return the exact same response for  both scenarios

They only have 2 options for subscribing to tickers
- subscribing to one market's tickers
- subscribing to all markets tickers

so we wouldn't know which message hash it's for

Reply:
@samgermain 
> There's no way to do that

Yes there is, you just need to create proper messageHashes and handle things correctly inside handleTickers, check for instance how Pablo did it here: https://github.com/ccxt/ccxt/pull/18023

Reply:
@samgermain I'm not talking about the subscription part, ofc you will need to subscribe to every ticker in this case, but upon receiving them you can always filter them out and resolve only the correct futures. If you have a future with a message hash "tickers::BTC/USDT,LTC/USDT" after getting an update with every ticker, you take from them only LTC/USDT and BTC/USDT and resolve that future, as I said check how Pablo did it, you can do the same thing here.